### PR TITLE
Adds saml2 logout

### PR DESCRIPTION
**Pull request by Frodotus**

### CODE REVIEW

**Review by suutari-ai — COMMENTED** [review]
Should it still logout locally even in the error cases?

Is it even logging out the local session in all cases? This doesn't seem clear at all when reading the hairy if-else code of the new `signout` function.

**Comment by Frodotus** [comment]
I think it should not logout locally in error cases. If it logs out locally, user then can't try to logout again.
Local logout should not be called before Idp confirms logout and local logout is not even wanted in all cases.

File: django_saml2_auth/views.py [hunk 1]
```diff
@@ -249,5 +275,33 @@ def signin(r):
...
+                body = ''.join(http_info['data'])
+                return HttpResponse(body)
+            elif binding == BINDING_HTTP_REDIRECT:
+                return HttpResponseRedirect(_get_location(http_info))
+            else:
+                return HttpResponseServerError('Failed to log out')
+        else:
+            return finish_signout(r, logout_info)
+
+    return HttpResponseServerError('Logout Binding not supported')
```
Comment (suutari-ai) [inline django_saml2_auth/views.py]: Add comments here to the if-else cases or to the function docstring to help the reader of the code to understand it. Currently the code raises the following questions, which IMHO should be answered some how:
 * What does it mean if logout_info is tuple or not? (In what cases it is and when not?)
 * What logout_info is, if it's not a tuple? (type and content)
 * What http_info contains?
 * What are the cases that are being handled when binding is POST or REDIRECT and why they are handled the way they are?
 * Why other bindings are raising a failure?
 * Why `finish_signout` is not always called?
 * Why user is not logged out locally in most of the cases?

File: django_saml2_auth/views.py [hunk 2]
```diff
@@ -249,5 +275,58 @@ def signin(r):
...
+                return HttpResponse(body)
+            elif binding == BINDING_HTTP_REDIRECT:
+                # Redirect to address defined in logout response
+                return HttpResponseRedirect(_get_location(http_info))
+            else:
+                # Unknown binding
+                return HttpResponseServerError('Logout binding not supported')
+        else:
+            # Soap logout does not return tupple and does not call callback so we logout locally here
+            return finish_signout(r, logout_info)
```
Comment (suutari-ai) [inline django_saml2_auth/views.py]: What is "Soap logout"?

You should also test all these branches. This at least is using a parameter that is undefined.

Comment (Frodotus) [reply django_saml2_auth/views.py]: Removed this as we can't test it.

File: django_saml2_auth/urls.py [hunk 3]
```diff
@@ -7,4 +7,5 @@
     url(r'^acs/$', views.acs, name="acs"),
     url(r'^welcome/$', views.welcome, name="welcome"),
     url(r'^denied/$', views.denied, name="denied"),
+    url(r'^finish-logout/$', views.finish_signout, name="denied"),
```
Comment (suutari-ai) [inline django_saml2_auth/urls.py]: name shouldn't be copy-pasted like that :)

Comment (Frodotus) [reply django_saml2_auth/urls.py]: 🤦‍♂️

**Review by RauliL — APPROVED** [review]
Because I am not a SAML expert, we went through this step by step in Slack where @Frodotus explained to me what this is supposed to do and why. It looks fine and dandy to me now.

**Review by suutari-ai — APPROVED** [review]
:+1:

**Pull request merged**